### DISCUSSION TITLE
How can I format date that was stored in properties ?

### DISCUSSION THREAD

**Discussion by ezekel** [spatie/laravel-activitylog#758] [Q&A]
Need some help to please, how can I format the `expense_date` to Y-m-d when storing to activity_log or when retrieving?

I tried to create Activity model and extends Spatie\Activitylog\Models\Activity as SpatieActivity;

I have this method helper , still I cannot achieve what I want.

Thank you in advance

```
 protected $casts = [
        'paid_at' => 'date:Y-m-d',
        'expense_date' => 'date:Y-m-d',
    ];


    protected function serializeDate(DateTimeInterface $date)
    {
        return $date->format('Y-m-d');
    }`

```

```
App\Activity {#4323
         id: 44,
         log_name: "Expense",
         description: "created",
         subject_id: 266,
         subject_type: "App\Expense",
         causer_id: 1,
         causer_type: "App\User",
         properties: "{"attributes":{"reason":"test create","created_by":1,"expense_date":"2020-07-26T00:00:00.000000Z","group_id":1,"lender":2,"borrower":3,"acctterm_id":2,"status_id":8,"category_id":3,"amount":null,"paid_date":null}}",
         created_at: "2020-07-26 11:17:00",
         updated_at: "2020-07-26 11:17:00",
       },
```

**Comment by Gummibeer** [comment]
Hey,

If `expense_date` is a datetime/timestamp in the original/source table you shouldn't modify it on save in the `properties` column but use carbon to format it during render.

But if you really want to change the stored value you can use the `tap()` or `tapActivity()` method or an observer. Whichever you like more and matches your exact usecase.

**Reply by ezekel** [reply]
Ok I will not modify when storing but is there  a way to automatically format when I retrieve the Activiy::find(3)->last() ? I cannot format or use carbon in vue templates

Thank you
